pkg/controllers/gacha-controller.go

```diff
@@ -141,59 +143,46 @@ func HandleGachaDraw(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "Internal Server Error", http.StatusInternalServerError)
 		return
 	}
-	characterPool := generateCharacterPool(characters)
+	characterPool := GenerateCharacterPool(characters)
 	response := models.GachaDrawResponse{
 		Results: make([]models.CharacterResponse, reqBody.Times),
 	}
+	// Create a slice to store the user characters for batch insert
+	userCharacters := make([]*models.UserCharacter, reqBody.Times)
 
-	// A buffered channel to receive the drawn characters
-	drawnCharacters := make(chan models.CharacterResponse, reqBody.Times)
-
-	//  WaitGroup to synchronize goroutines
-	var wg sync.WaitGroup
-
-	// Worker pool with a fixed number of goroutines
-	numWorkers := 10
-	tasks := make(chan int, reqBody.Times)
-
-	// Start goroutines to process tasks concurrently
-	for i := 0; i < numWorkers; i++ {
-		wg.Add(1)
-		go func() {
-			defer wg.Done()
-			for range tasks {
-				character, err := models.DrawCharacter(characters, characterPool) // Simulate drawing a character
-				if err != nil {
-					http.Error(w, "Internal Server Error", http.StatusInternalServerError)
-					return
-				}
-				characterResponse := models.CharacterResponse{
-					CharacterID: fmt.Sprintf("Character-%d", character.ID),
-					Name:        character.Name,
-					Rarity:      character.Rarity,
-				}
-				drawnCharacters <- characterResponse
-			}
-		}()
-	}
-
-	// Enqueue tasks
 	for i := 0; i < reqBody.Times; i++ {
-		tasks <- i
-	}
-
-	// Close the tasks channel and wait for all goroutines to finish
-	go func() {
-		close(tasks)
-		wg.Wait()
-		close(drawnCharacters)
-	}()
+		character, err := DrawCharacter(characterPool)
+		if err != nil {
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+			return
+		}
+		userCharacter := models.UserCharacter{
+			UserID:            user.ID,
+			CharacterID:       character.ID,
+			AttackPower:       character.AttackPower,
+			Defense:           character.Defense,
+			Speed:             character.Speed,
+			HitPoints:         character.HitPoints,
+			CriticalHitRate:   character.CriticalHitRate,
+			ElementalAffinity: character.ElementalAffinity,
+			Rarity:            character.Rarity,
+			Synergy:           character.Synergy,
+			Evolution:         character.Evolution,
+		}
+		// Store the user character in the batch slice
+		userCharacters[i] = &userCharacter
 
-	// Collect the drawn characters from the channel and populate the response
-	i := 0
-	for characterResponse := range drawnCharacters {
+		characterResponse := models.CharacterResponse{
+			CharacterID: fmt.Sprintf("Character-%d", character.ID),
+			Name:        character.Name,
+			Rarity:      character.Rarity,
+		}
 		response.Results[i] = characterResponse
-		i++
+	}
+	// Batch insert user characters into the database
+	if err := models.CreateUserCharacterBatch(userCharacters); err != nil {
```

Comment:
Please name it in a way that makes it clear that you are not creating a batch, but rather creating characters with batch.

Reply:
Okay I meant create user's character batch hence I named it that way.However I will change it if it doesn't make understandable.